content/en/docs/advanced-tasks/adding-admission-and-conversion-webhooks.md

```diff
@@ -0,0 +1,138 @@
+- The `DeploymentName` that OLM must mount the CA Cert information into, this should match the name of a deployment defined in the CSV
+
+## Deploying an operator with webhooks using OLM
+
+### OLM Requirements for Operators with Webhooks
+
+OLM has a few constraints that must be considered when developing an operator featuring validating, mutating, or conversion webhooks.
+
+#### Certificate Authority Requirements
+
+OLM is configured to create a single Certificate Authority (CA) for each webhook deployment. The logic that generates and mounts the CA into the deployment was originally used by the [API Service](https://kubernetes.io/docs/tasks/access-kubernetes-api/setup-extension-api-server/) lifecycle logic. As a result:
+
+- The tls Cert file will be mounted to the deployment at `/apiserver.local.config/certificates/apiserver.crt`
+- The tls Key file will be mounted to the deployment at `/apiserver.local.config/certificates/apiserver.key`
+
+If the operator was built with [Kubebuilder](https://book.kubebuilder.io/) or the [Operator-SDK](https://github.com/operator-framework/operator-sdk), [this commit](https://github.com/awgreene/webhook-operator/commit/3ef33a1b7090180bd873fb2ecbcddb9a92a79f0b) can be used as a reference to update where the operator's webhook expects the Certs mounted by OLM.
+
+#### Admission Webhook Rule Requirements
+
+When developing an admission webhook please be aware that in an attempt to prevent operator from configuring the cluster into an unrecoverable state, OLM will place the CSV in the failed phase if the Rules defined in an admission webhook:
+
+- Intercept requests that target all groups
+- Intercept requests that target the `operators.coreos.com` group
+- Intercept requests that target the `ValidatingWebhookConfigurations` or `MutatingWebhookConfigurations` resources
+
+#### Conversion Webhook Rules Requirements
+
+- CSVs featuring a conversion webhook may only support only support the `AllNamespaces` installMode
+- The CRD targeted by the Conversion webhook must have it's `spec.preserveUnknownFields` field set to `false` or `nil`
+- The Conversion Webhook defined in the CSV must target an owned CRD
+
+### Example: Installing a operator featuring webhooks with OLM
+
+This example will focus on installing an operator that the OLM team built for it's e2e test portfolio, the [webhook-operator](https://github.com/awgreene/webhook-operator/tree/olm). The webhook-operator implements a validating, mutating, and conversion webhook for the [WebhookTest CRD](https://github.com/awgreene/webhook-operator/blob/olm/config/crd/bases/webhook.operators.coreos.io_webhooktests.yaml).
+
+#### Defining Conversion Webhooks in the ClusterServiceVersion
+
+The full webhook operator CSV can be seen [here](https://github.com/awgreene/webhook-operator/blob/07375018e32cbebedee55aa1b131469c247dd275/bundle/manifests/webhook-operator.clusterserviceversion.yaml), but we will highlight the important parts of the CSV below:
```

Comment:
Nit: I wonder if we can move an example yaml somewhere more visible/official than an individual repo. I guess it doesn't matter given you include a piece of it here.

Reply:
We will eventually update the location of that code to something more official.